src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java

```diff
@@ -459,6 +470,32 @@ private void checkTag(DetailAST ast, List<JavadocTag> tags, String tagName,
         }
     }
 
+    /**
+     * Verifies that a record definition has the specified param tag for
+     * the specified record component name.
+     *
+     * @param ast           the AST node for the record definition.
+     * @param tags          tags from the Javadoc comment for the record definition.
+     * @param recordComponentName the name of the type parameter
+     */
+    private void checkComponentParamTag(final DetailAST ast,
+                                        final List<JavadocTag> tags,
+                                        final String recordComponentName) {
+        boolean found = false;
+        for (int i = tags.size() - 1; i >= 0; i--) {
```

Comment:
Is the order of this loop important? (just curious)

Reply:
this comes from https://github.com/checkstyle/checkstyle/commit/9ce0ceb04ae90185f95643fb4484e525bcb0d3df
In 2004, this was an optimization. Now for the JIT compiler there is no difference between
`for (int i = 0; i < tags.size(); i++)`
and
`for (int i = tags.size() - 1; i >= 0; i--)`
and even
`for (final JavadocTag tag : tags)`

Reply:
@nmancus1 , lets make code more readable, please use "foreach"

Reply:
Fancy syntax added.